database/migrations/2017_03_25_043016_CreateNotificationsTable.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+use Illuminate\Support\Facades\Schema;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Database\Migrations\Migration;
+
+class CreateNotificationsTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('notifications', function (Blueprint $table) {
+            $table->increments('id');
+            $table->string('recipient_username');
+            $table->string('sender_username');
+            $table->smallInteger('category');
+            $table->smallInteger('type');
+            $table->string('parameters')->nullable();
+            $table->boolean('is_read')->default(0);
+            $table->integer('reference_id');
+            $table->timestamp('date_added');
```

Comment:
Friend request is created >> Notification generated (Category:Friend; Type:Request) 
In this scenario, the reference_id is the id from newly created Friend. The parameter is nullable as not every type of notification needs to pass extra information. For the friend scenario, the parameter could be used when a Friend request is accepted/denied where $notification->parameter ='accepted' or 'denied'.

We have Expense category too and parameters would hold the amount owed in an expense request for example.